site/docs/hosting/vercel.md

```diff
@@ -0,0 +1,117 @@
+
+## Configuring Vercel
+
+The next step is to create a `vercel.json` file at the top level of your project.
+For our example structure, its content would be:
+
+```json
+{
+  "functions": {
+    "api/bot.ts": {
+      "memory": 1024,
+      "maxDuration": 10
+    }
+  }
+}
+```
+
+> If you want to use Vercel's free subscription, your `memory` and `maxDuration` configurations might look like above to not bypass its limits.
+
+If you want to learn more about the `vercel.json` configuration file, see [its documentation](https://vercel.com/docs/project-configuration).
+
+## Configuring TypeScript
+
+In our `tsconfig.json`, we have to specify our output directory as `build/`, and our root directory as `api/`.
+This is important since we will specify them in Vercel's deploy options.
+
+```json{5,8}
+{
+  "compilerOptions": {
+    "target": "ES2019",
+    "module": "commonjs",
+    "rootDir": "./api",
+    "moduleResolution": "node",
+    "resolveJsonModule": true,
+    "outDir": "./build",
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "strict": true,
+    "skipLibCheck": true
+  }
+}
+```
+
+## The Main File
+
+Regardless of using TypeScript or JavaScript, we should have a source file through which our bot runs.
+It should look roughly like this:
+
+```ts
+import { Bot, webhookCallback } from "grammy";
+
+const bot = new Bot(process.env.BOT_TOKEN); // <-- insert your bot token from environment variable
+
+export default webhookCallback(bot, "http"); // <-- This line will do the trick!
+```
+
+## Go to Vercel Website
+
+Assuming you have a Vercel account and have your GitHub account connected to Vercel, add a new project and select your GitHub bot repository.
+In the _Build & Development Settings_:
+
+- Output directory: `build`
+- Install command: `npm install`
+
+Don't forget to add **environment variables** such as your **bot token** in the environment variables inside Vercel settings. Once you have done, deploy it!
+
+## Setting Up the Telegram Webhook
+
+Your bot is ready.
+The last step is to add the webhook that connects your deployment with Telegram and your bot token.
+Here is the URL you need to visit using your browser:
+
+```md:no-line-numbers
+https://api.telegram.org/bot<BOT_TOKEN>/setWebhook?url=<HOST_URL>
+```
+
+The `HOST_URL` is a little tricky, because you need to use your **Vercel app domain following with the route to the bot code**, for example `https://appname.vercel.app/api/bot`.
+Where `bot` is your `bot.ts` or `bot.js` file.
+
+If everything was successful, you should now see this in your browser window.
+
+```json
+{
+  "ok": true,
+  "result": true,
+  "description": "Webhook was set"
+}
+```
+
+Other useful commands are:
+
+- Info about webhook: `https://api.telegram.org/bot{BOT_TOKEN}/getWebhookInfo`
+- Delete webhook: `https://api.telegram.org/bot{BOT_TOKEN}/deleteWebhook`
```

Comment:
I think that this is more than necessary since we have not included it in the other guides, and we’re working on #523.

Reply:
Ok, so, should I delete it?

Reply:
Yes, it would be way better.

Reply:
As soon as this PQ is done and merged, we should probably copy this paragraph to the other guides, as well. Not very DRY, but very useful indeed.

Reply:
Done!